ACP2E-4454: [Documentation] Update devdocs how to keep inheritance of product gallery in REST API when updating a product in store view level. (#548)
AdobeDocs/commerce-webapi: src/pages/rest/tutorials/image/product-image-store-scope.md

**Endpoint:**

```html
POST http://domain.com/rest/default/V1/products
```

<CodeBlock slots="heading, code" repeat="2" languages="JSON, JSON" />

#### Payload

```json
{
  "product": {
    "sku": "MJ03",
    "name": "product name override in default store view",
    "media_gallery_entries": null
  }
}
```

#### Response

```json
{
  "sku": "MJ03",
  "name": "product name override in default store view",
  "media_gallery_entries": [
    {
      "id": 709,
      "media_type": "image",
      "label": "",
      "position": 1,
      "disabled": false,
      "types": [
        "image",
        "small_image",
        "thumbnail"
      ],
      "file": "\/m\/j\/mj03-black_main_12.jpg"
    },
    {
      "id": 710,
      "media_type": "image",
      "label": "",
      "position": 2,
      "disabled": false,
      "types": [],
      "file": "\/m\/j\/mj03-black_alt1_12.jpg"
    },
    {
      "id": 711,
      "media_type": "image",
      "label": "",
      "position": 3,
      "disabled": false,
      "types": [],
      "file": "\/m\/j\/mj03-black_back_12.jpg"
    }
  ]
}
```

> 

Author: ```suggestion

```

---
title: Manage product images at the store view level
description: Learn how to use the Adobe Commerce REST API to preserve, override, and restore product image gallery settings per store view in a multi-store setup.
keywords:
  - REST
---

# Manage product images at the store view level

> This page explains how product image gallery inheritance works in a multi-store Adobe Commerce setup and how to control it using the REST API. These behaviours only apply when your installation has multiple store views. In a single-store setup, all requests target the global scope and inheritance does not apply.

Reviewer: ```suggestion
This topic explains how product image gallery inheritance works in a multi-store Adobe Commerce setup and how to control it using the REST API. These behaviours only apply when your installation has multiple store views. In a single-store setup, all requests target the global scope and inheritance does not apply.
```

**Endpoint:**

```html
POST http://domain.com/rest/default/V1/products
```

<CodeBlock slots="heading, code" repeat="2" languages="JSON, JSON" />

#### Payload

```json
{
  "product": {
    "sku": "MJ03",
    "name": "product name override in default store view"
  }
}
```

#### Response

```json
{
  "sku": "MJ03",
  "name": "product name override in default store view",
  "media_gallery_entries": [
    {
      "id": 709,
      "media_type": "image",
      "label": "",
      "position": 1,
      "disabled": false,
      "types": [
        "image",
        "small_image",
        "thumbnail"
      ],
      "file": "\/m\/j\/mj03-black_main_12.jpg"
    },
    {
      "id": 710,
      "media_type": "image",
      "label": "",
      "position": 2,
      "disabled": false,
      "types": [],
      "file": "\/m\/j\/mj03-black_alt1_12.jpg"
    },
    {
      "id": 711,
      "media_type": "image",
      "label": "",
      "position": 3,
      "disabled": false,
      "types": [],
      "file": "\/m\/j\/mj03-black_back_12.jpg"
    }
  ]
}
```

> After this call, images in each store view continue to use the **Use Default Value** setting.

Reviewer: ```suggestion
After this call, images in each store view continue to implement the **Use Default Value** setting.
```

### Set media_gallery_entries to null

> Explicitly setting `media_gallery_entries` to `null` has the same effect as omitting the field — the product's existing media gallery is left unchanged.

Reviewer: ```suggestion
Setting `media_gallery_entries` to `null` and omitting the field are equivalent. Neither modifies the images currently configured for any store view.
```

> 

Author: ```suggestion
Setting `media_gallery_entries` to `null` and omitting the field are equivalent. Neither modifies the images currently configured for any store view.

<InlineAlert variant="warning" slots="text"/>

Setting `media_gallery_entries` to an empty array `[]` is **not** the same as `null`. An empty array is treated as an explicit instruction to remove all gallery entries from all store views. Always use `null` or omit the field when you do not intend to modify images.

```

**Endpoint:**

```html
POST http://domain.com/rest/default/V1/products
```

<CodeBlock slots="heading, code" repeat="2" languages="JSON, JSON" />

#### Payload

```json
{
  "product": {
    "sku": "MJ03",
    "name": "product name override in default store view",
    "media_gallery_entries": null
  }
}
```

#### Response

```json
{
  "sku": "MJ03",
  "name": "product name override in default store view",
  "media_gallery_entries": [
    {
      "id": 709,
      "media_type": "image",
      "label": "",
      "position": 1,
      "disabled": false,
      "types": [
        "image",
        "small_image",
        "thumbnail"
      ],
      "file": "\/m\/j\/mj03-black_main_12.jpg"
    },
    {
      "id": 710,
      "media_type": "image",
      "label": "",
      "position": 2,
      "disabled": false,
      "types": [],
      "file": "\/m\/j\/mj03-black_alt1_12.jpg"
    },
    {
      "id": 711,
      "media_type": "image",
      "label": "",
      "position": 3,
      "disabled": false,
      "types": [],
      "file": "\/m\/j\/mj03-black_back_12.jpg"
    }
  ]
}
```

> <InlineAlert variant="info" slots="text"/>

Reviewer: Delete this inline alert and paragraph. I replaced the text in line 97 with the text listed here.

> Setting `media_gallery_entries` to `null` and omitting the field are equivalent. Neither modifies the images currently configured for any store view.

Author: ```suggestion
```

> <InlineAlert variant="warning" slots="text"/>

Reviewer: Move this InlineAlert and paragraph to be before the `**Endpoint**` line currently on line 99

Setting `media_gallery_entries` to an empty array `[]` is **not** the same as `null`. An empty array is treated as an explicit instruction to remove all gallery entries from all store views. Always use `null` or omit the field when you do not intend to modify images.

> 

Author: ```suggestion
```

## Override store view image fields

When you retrieve a product and POST it back with the `media_gallery_entries` array populated, the store view fields are explicitly set — overriding any inherited values.

**Step 1.** Retrieve the full product data from the store view you want to update.

**Endpoint:**

```html
GET http://domain.com/rest/default/V1/products/MJ03
```

**Step 2.** Use the `media_gallery_entries` from the GET response in an update call.

**Endpoint:**

```html
POST http://domain.com/rest/default/V1/products
```

After this call:

*  In the **default store view**: image sort order, label, visibility, and roles are no longer using the default value (**Use Default Value** is unchecked).

*  In the **second store view**: image fields continue to use the default value (unaffected).

## Restore inheritance by setting fields to null

> Setting individual image fields to `null` restores the **Use Default Value** state for those fields — equivalent to re-enabling the **Use Default Value** checkbox in the Admin product editor.

Reviewer: ```suggestion
Setting individual image fields to `null` restores the **Use Default Value** state for those fields. This is the equivalent of re-enabling the **Use Default Value** checkbox in the Admin product editor.
```
